src/gentropy/dataset/study_index.py

```diff
@@ -590,3 +592,158 @@ def annotate_sumstats_qc(
+            Window.unboundedPreceding, Window.unboundedFollowing
+        )
+
+        # Temporary columns to drop at the end:
+        columns_to_drop = ["keepTopHit", "mostGranular", "rank"]
+
+        return StudyIndex(
+            _df=(
+                self.df
+                # Initialising quality controls column, if not present:
+                .withColumn(
+                    "qualityControls",
+                    f.when(
+                        f.col("qualityControls").isNull(),
+                        f.array().cast(ArrayType(StringType())),
+                    ).otherwise(f.col("qualityControls")),
+                )
+                # Keeping top hit studies unless the same study is available from a summmary statistics source:
+                # This value will be set for all rows for the same `studyId`:
+                .withColumn(
+                    "keepTopHit",
+                    f.when(
+                        f.array_contains(
+                            f.collect_set(f.col("hasSumstats")).over(
+                                full_study_id_window
+                            ),
+                            True,
+                        ),
+                        f.lit(False),
+                    ).otherwise(True),
+                )
+                # For studies without summary statistics, we remove the "Not curated by Open Targets" flag:
+                .withColumn(
+                    "qualityControls",
+                    f.when(
+                        ~f.col("hasSumstats"),
+                        f.array_remove(
+                            f.col("qualityControls"),
+                            StudyQualityCheck.NO_OT_CURATION.value,
+                        ),
+                    ).otherwise(f.col("qualityControls")),
+                )
+                # If top hits are not kept, we remove the "sumstats not available" flag from all QC lists:
+                .withColumn(
+                    "qualityControls",
+                    f.when(
+                        ~f.col("keepTopHit"),
+                        f.array_remove(
+                            f.col("qualityControls"),
+                            StudyQualityCheck.SUMSTATS_NOT_AVAILABLE.value,
+                        ),
+                    ).otherwise(f.col("qualityControls")),
+                )
+                # Then propagate quality checks for all sources of the same study:
+                .withColumn(
+                    "qualityControls",
+                    f.array_distinct(
+                        f.flatten(
+                            f.collect_set("qualityControls").over(full_study_id_window)
+                        )
+                    ),
+                )
+                # Propagating sumstatQCValues -> map, cannot be flatten:
+                .withColumn(
+                    "sumstatQCValues",
+                    f.first("sumstatQCValues", ignorenulls=True).over(
```

Comment:
There always should be only one row with `sumstatQCValues` populated. otherwise the study will be flagged out as invalid right?